pkg/logentry/pipeline.go

```diff
@@ -7,24 +7,34 @@ import (
 	"github.com/go-kit/kit/log/level"
 	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 	"github.com/prometheus/common/model"
 
 	"github.com/grafana/loki/pkg/logentry/stages"
 	"github.com/grafana/loki/pkg/promtail/api"
 )
 
+var (
+	pipelineDuration = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Namespace: "logentry",
+		Name:      "pipeline_duration_seconds",
+		Help:      "Label and metric extraction pipeline processing time, in seconds",
+		Buckets:   []float64{.000005, .000010, .000025, .000050, .000100, .000250, .000500, .001000, .002500, .005000, .010000, .025000},
```

Comment:
prometheus.LinearBuckets or prometheus.ExponentialBuckets

Reply:
would be a little more concise, but for my brain I find it easier to just see the values, especially when dealing with microseconds